### ISSUE TITLE
Gradient mixins don't handle the transparent keyword.

### ISSUE THREAD

**Issue by chriseppstein** [Compass/compass#356]
it should treat it as `rgba(#000, 0)` for the purpose of calculations but output as transparent.

**Comment by eklipse2k8** [comment]
I'm hitting this :-/.

**Comment by NoNoNo** [comment]
Me too, when testing the effect described in http://lea.verou.me/demos/css3-patterns.html

\+ When SVG-support is on, the Mixin `background-image` emits a `background-size: 100%`, which will overwrite my declaration… :(

**Comment by ryanlindsey** [comment]
I can't compile Twitter's Bootstrap in production mode due to this error...

**Comment by ngs** [comment]
I wrote a patch. https://gist.github.com/1722819

**Comment by yhbrandon** [comment]
how do you apply your patch?

**Comment by scottdavis** [comment]
@ngs mind making a pull request?

**Comment by alanhogan** [comment]
Stupid question / #lazyweb but:
- Is `transparent` really the same as `rgba(0,0,0 0.0)`?
- E.g. if a gradient is made from both of these values to `white` (`rgba(255,255,255, 1.0)`), do they look the same?

I would sort of naïvely expect that fading from transparent to white would involve fading through e.g. `rgba(255,255,255, 0.5)` et al, but that fading from zero-alpha black to white would mean fading through `rgba(127,127,127, 0.5)` which would look different / grayer / darker.

**Comment by chriseppstein** [comment]
I don't think it's the same, which is why the patch I made simply passes the transparent keyword through to the generated css. It was more code, but I think it's better to stay true to the input.

**Comment by ryanlindsey** [comment]
Replacing transparent with rgba values isn't the same, but it is close. [Here's a fiddle](http://jsfiddle.net/pixelsonly/4kWFS/)

**Comment by chriseppstein** [comment]
yep. that's what I expected it would do. thank you for confirming.

**Comment by alanhogan** [comment]
fantastic, thanks for doing this the Right Way, Chris.

**Comment by chriseppstein** [comment]
![Fuck Yea](http://markreads.net/reviews/wp-content/uploads/2011/03/FUCK-YEA.png)

**Comment by jDeppen** [comment]
According to the changelog, this was fixed in 0.11.8 (02/26/2012) but that gem isn't available

```
ERROR:  Could not find a valid gem 'compass' (= 0.11.8) in any repository
```

I'm using Rails 2.3 trying to add bootstrap

http://compass-style.org/CHANGELOG/
0.11.8 (02/26/2012)
Fix a bug in gradients that used the transparent keyword

I added gradient_support.rb to config/initializers and it fixed the issue
https://gist.github.com/1722819
